apps/api/src/migration/1663167395345-ChangeJobTablePKtoUUID.ts

```diff
@@ -0,0 +1,223 @@
+    await queryRunner.query(`
+        ALTER TABLE "${jobLocationTable}" DROP CONSTRAINT "FK_4e8a99227d12fd8e7988b7d5b88"
+    `);
+
+    await queryRunner.query(`
+        ALTER TABLE "${jobLocationTable}" DROP COLUMN "ien_applicant_jobs_id";
+    `);
+
+    await queryRunner.query(`
+        ALTER TABLE "${jobLocationTable}" RENAME COLUMN "temp_uid" TO "ien_applicant_jobs_id";
+    `);
+
+    await queryRunner.dropPrimaryKey(jobTable);
+    await queryRunner.dropColumn(jobTable, 'id');
+
+    await queryRunner.changeColumn(
+      jobTable,
+      'uid',
+      new TableColumn({
+        type: 'uuid',
+        name: 'id',
+        isPrimary: true,
+        isGenerated: true,
+        generationStrategy: 'uuid',
+      }),
+    );
+
+    await queryRunner.query(`
+      ALTER TABLE public.${statusAuditTable} ADD CONSTRAINT "FK_ec2940cc497a06ba811763be464" FOREIGN KEY (job_id) REFERENCES public.${jobTable}(id) ON DELETE CASCADE ON UPDATE NO ACTION;
+    `);
+
+    await queryRunner.query(`
+      ALTER TABLE public.${jobLocationTable} ADD CONSTRAINT "FK_4e8a99227d12fd8e7988b7d5b88" FOREIGN KEY (ien_applicant_jobs_id) REFERENCES public.${jobTable}(id) ON DELETE CASCADE ON UPDATE NO ACTION;
+    `);
+
+    await queryRunner.query(`
+      CREATE UNIQUE INDEX "unique_applicant_status_date_job"
+      ON "ien_applicant_status_audit" (applicant_id, status_id, start_date, job_id)
+      WHERE job_id IS NOT NULL
+    `);
+  }
+
+  public async down(queryRunner: QueryRunner): Promise<void> {
+    const jobTable = 'ien_applicant_jobs';
+    const statusAuditTable = 'ien_applicant_status_audit';
+    const jobLocationTable = 'ien_applicant_jobs_job_location_ien_job_locations';
+    let count = 0;
+
+    // alter status audit table
+    await queryRunner.query(`
+      ALTER TABLE "${jobTable}" ADD COLUMN IF NOT EXISTS "temp_id" integer DEFAULT NULL;
+    `);
+    const jobsToUpdate: { id: string }[] = await queryRunner.query(`
+      SELECT id FROM "${jobTable}"
+    `);
+    // map existing jobs with ids to uid column from ien_applicant_jobs
+    await Promise.all(
+      jobsToUpdate.map(({ id }) => {
+        return queryRunner.query(`
+        UPDATE "${jobTable}" j SET "temp_id" = ${count++} 
+        WHERE j.id = '${id}
+      `);
+      }),
+    );
+
+    const oldJobs: { id: string; job_id: string }[] = await queryRunner.query(
+      `SELECT id, job_id FROM "${statusAuditTable}" WHERE job_id IS NOT NULL`,
+    );
+    // alter status audit table
+    await queryRunner.query(`
+      ALTER TABLE "${statusAuditTable}" ADD COLUMN IF NOT EXISTS "temp_id" integer DEFAULT NULL;
+    `);
+    // map existing jobs with ids to uid column from ien_applicant_jobs
+    await Promise.all(
+      oldJobs
+        .filter(job => job.job_id)
+        .map(job => {
+          return queryRunner.query(`
+            UPDATE "${statusAuditTable}" s SET "temp_uid" = (SELECT temp_id FROM "${jobTable}" j WHERE j.id = '${job.job_id}')
```

Comment:
I hope we'd never run this but should `temp_uid` be `temp_id`?

Reply:
you are correct, nice catch